core/vm/message/impl/message_signer_impl.hpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_VM_MESSAGE_SIGNER_IMPL_HPP
+#define CPP_FILECOIN_CORE_VM_MESSAGE_SIGNER_IMPL_HPP
+
+#include "common/logger.hpp"
+#include "storage/keystore/keystore.hpp"
+#include "vm/message/message_signer.hpp"
+
+namespace fc::vm::message {
+
+  using storage::keystore::KeyStore;
+
+  /**
+   * Class MessageSignerImpl.
+   */
+  class MessageSignerImpl : public MessageSigner {
+   public:
+    MessageSignerImpl() = delete;
+    explicit MessageSignerImpl(std::shared_ptr<KeyStore> ks);
+
+    /**
+     * @copydoc MessageSigner::sign()
+     */
+    outcome::result<SignedMessage> sign(
+        const Address &address, const UnsignedMessage &msg) noexcept override;
+
+    /**
+     * @copydoc MessageSigner::verify()
+     */
+    outcome::result<UnsignedMessage> verify(
+        const Address &address, const SignedMessage &msg) noexcept override;
```

Comment:
```suggestion
        const Address &address, const SignedMessage &msg) const noexcept override;
```
It could be marked const